### ISSUE TITLE
NodePort service is not load balancing by source ip but by (source ip+destnation port of server)

### ISSUE THREAD

**Issue by rizwan3659** [projectcalico/calico#6281]
I have migrated IMS (IP Multimedia Service ) as a service on kubernetes with Calico Cni . IMS node has a pcscf server which is running as a NodePort service. For this service, 3 pcscf pods are running. pcscf server has opened 2 ports (5060, 40982 )to listen. There is requirement where client sends first SIP (layer 7 ) message to 5060 at service ip of pcscf and 2nd SIP message to 40982 at service ip of pcscf. The problem I am facing that. First SIP message is going to first pod and 2nd SIP message is going to another pod. I want load balancing based on source ip of client and not on (source ip + destination port of pcscf). Any help will be appreciated

## Expected Behavior
I want load balancing by source ip and not by source ip + destination port of server.

**Comment by caseydavenport** [comment]
@rizwan3659 could you please include whether you're using Calico in iptables mode or in eBPF mode? 

In iptabales mode, Calico is not responsible for service load balancing, but Calico does perform service LB for eBPF mode.

**Comment by rizwan3659** [comment]
> @rizwan3659 could you please include whether you're using Calico in iptables mode or in eBPF mode?
> 
> In iptabales mode, Calico is not responsible for service load balancing, but Calico does perform service LB for eBPF mode.

Thanks for the reply. Yes I am running calico in iptables mode.  I have tried also running kube-proxy in ipvs mode with rr as being the scheduler. I was able to load balance by source ip. 
I am facing following problems:-
1) 5% of the 1 lakh client sip messages did not go to correct pod( 2nd sip message goes to different pod). 
2) 2nd issue is that  service does not distribute the UDP packets among the multiple real servers (pods )in a round robin fashion equally . It  sends to multiple servers(pods) in unequal amount. The reason I think is that  IPVS udp_timeout whose minimum value is 1 second. SInce, UDP packets are continuously sent by the proxy to the service, the timeout never really expires.


I will be try calico in eBPF mode and will let you know.
